### PR TITLE
Lecture2: homework (convert JS to TS), add RestaurantsPage with splitting to components

### PR DESCRIPTION
**Pull request by ivan-education**
- convert JS project to TS project
- add RestaurantsPage with splitting to components
- add `absolute import paths` approach

### CODE REVIEW

File: src/components/reviews/component.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { ReviewEntity } from "src/types";
+
+export const Reviews: React.FC<{ reviews: ReviewEntity[] }> = ({ reviews }) => {
+  return (
+    <>
+      <h3>Reviews:</h3>
+      <ul>
+        {reviews.map((review: ReviewEntity) => {
+          return (
+            <li key={review.id}>
```
Comment (SArchieEdu) [inline src/components/reviews/component.tsx:11]: лучше в отдельный компонент

File: src/components/menu/component.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  dishes: DishEntity[];
+}
+
+export const Menu: React.FC<Props> = ({ dishes }) => {
+  return (
+    <>
+      <h3>Menu:</h3>
+      <ul>
+        {dishes.map((dish: DishEntity) => {
+          return (
```
Comment (SArchieEdu) [inline src/components/menu/component.tsx:14]: аналогично

**Review by SArchieEdu — APPROVED** [review]

File: src/components/reviews/component.tsx (hunk 1 above)
Comment (ivan-education) [reply src/components/reviews/component.tsx:11]: готово

File: src/components/menu/component.tsx (hunk 2 above)
Comment (ivan-education) [reply src/components/menu/component.tsx:14]: готово

**Pull request merged**